### ISSUE TITLE
Writing introductory article about internal process.

### ISSUE THREAD

**Issue by pocorall** [pfn/android-sdk-plugin#202]
The main part of `README.md` is basically a changelog. It is OK for just using this plugin. However, in case of facing trouble with build, there are no guide about what is going on inside of this plugin. Although the code is written pretty decently, not all part of the implementation is intuitive. So the document written in plain English would be helpful.

**Comment by charafau** [comment]
This is good idea. Maybe we could elaborate more on this? If you have some ideas please join our gitter room for further discussion  ( https://gitter.im/scala-android/sbt-android )
